builtin/mainmenu/dlg_config_world.lua

```diff
@@ -195,10 +198,16 @@ local function handle_buttons(this, fields)
 		for i,mod in ipairs(rawlist) do
 			if not mod.is_modpack and
 					mod.typ ~= "game_mod" then
-				if mod.enabled then
-					worldfile:set("load_mod_"..mod.name, "true")
+				if modname_valid(mod.name) then
+					if mod.enabled then
+						worldfile:set("load_mod_"..mod.name, "true")
+					else
+						worldfile:set("load_mod_"..mod.name, "false")
+					end
 				else
-					worldfile:set("load_mod_"..mod.name, "false")
+					if mod.enabled then
+						gamedata.errormessage = core.gettext("Failed to enable mod \"%s\" as it contains disallowed characters. Only chararacters [a-z0-9_] are allowed."):format(mod.name)
```

Comment:
The problem is that translation files will have a mix of %s and $1 if we use two different methods. Just stick with $1.